bundles/ui/org.eclipse.smarthome.ui.basic/src/main/java/org/eclipse/smarthome/ui/basic/internal/render/AbstractWidgetRenderer.java

```diff
@@ -0,0 +1,188 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.smarthome.ui.basic.internal.render;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.model.sitemap.Widget;
+import org.eclipse.smarthome.ui.basic.internal.WebAppActivator;
+import org.eclipse.smarthome.ui.basic.render.RenderException;
+import org.eclipse.smarthome.ui.basic.render.WidgetRenderer;
+import org.eclipse.smarthome.ui.items.ItemUIRegistry;
+import org.osgi.service.component.ComponentContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This is an abstract implementation of a widget renderer. It provides
+ * methods that are very useful for any widget renderer implementation,
+ * so it should be subclassed by most concrete implementations.
+ *
+ * @author Kai Kreuzer - Initial contribution and API
+ *
```

Comment:
please add yourself as a second author tag in every class that you have touched!